tests/Hilke.KineticConvolution.Tests/ConvolutionFactoryTests.cs

```diff
@@ -441,5 +454,118 @@ public void When_calling_Convolve_With_two_segment_with_opposite_orientation_The
             // Assert
             actual.Should().BeEmpty();
         }
+
+        [Test]
+        public void When_Two_Shapes_Are_Convolved_Then_Parent1_Should_Belong_To_Shap1_And_Parent2_Should_Belong_To_Shape2()
+        {
+            // Arrange
+            var shape1 = CreateShape1();
+            var shape2 = CreateShape2();
+
+            // Act
+            var convolution = _factory.ConvolveShapes(shape1, shape2);
+
+            // Assert
+            convolution.ConvolvedTracings.ToList()
+                .ForEach(convolvedTracing =>
+                    shape1.Tracings.Contains(convolvedTracing.Parent1).Should().BeTrue());
+
+            convolution.ConvolvedTracings.ToList()
+                .ForEach(convolvedTracing =>
+                    shape2.Tracings.Contains(convolvedTracing.Parent2).Should().BeTrue());
+
+            Shape<double> CreateShape1()
```

Comment:
If it's an internal function, I think that first letter should be lowercase

Reply:
Done